test/lib/utils.go

```diff
@@ -32,3 +42,27 @@ func GetEnvOrFail(t *testing.T, key string) string {
 	}
 	return value
 }
+
+// WaitForSourceAuthCheckPending polls the status of the Source from client
+// every interval until isSourceAuthCheckPending returns `true` indicating
+// it is done, or returns an error, or timeout.
+func WaitForSourceAuthCheckPending(dynamicClient dynamic.Interface, obj *resources.MetaResource) error {
```

Comment:
AuthCheckPending is a state that will continue until the auth check succeeds, right? If so, it is worth commenting here to that effect. Namely that there is no race condition with the auth check finishing with a failure because Pending is the 'has not yet succeeded' state.

Reply:
One thing to be noted is AuthCheckPending state will be momently turned back to Ready when Pod restarts (at the beginning of a Pod restarts, Pod state is Ready before auth check running in the Pod is performed.) That's why I use the waitForSourceAuthCheckPending function instead of directly sleeping 2 min, and getting the Source state to verify if it is in AuthCheckPending (it may in its temporarily Ready state)